src/App.jsx

```diff
@@ -1,14 +1,42 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import data from './data/messages.json';
 
 const App = () => {
+  const [entryData, setEntryData] = useState(data);
+
+  const handleLikeStatus = (id) => {
+    setEntryData(entryData => entryData.map(entry => {
+      if (entry.id === id) {
+        return {...entry, liked: !entry.liked};
+      } else {
+        return entry;
+      }
+    }));
+  };
+
+  const calculateTotalLikes = (entryData) => {
+    return entryData.reduce((total, entry) => {
+      return total + entry.liked;
+    }, 0);
+  };
```

Comment:
Nice function, I would say the only thing here is that you don't need to pass anything into since you will always be working with `entryData` here. 
```suggestion
  const calculateTotalLikes = () => {
    return entryData.reduce((total, entry) => {
      return total + entry.liked;
    }, 0);
  };
```